src/App.tsx

```diff
@@ -157,239 +286,541 @@ function App() {
+                                scope as keyof typeof FEATURED_NETWORKS
+                              ]
+                            } (${scope})`
+                          : scope
+                      }
+                      className="scope-card-title"
+                    >
+                      {FEATURED_NETWORKS[
+                        scope as keyof typeof FEATURED_NETWORKS
+                      ]
+                        ? `${
+                            FEATURED_NETWORKS[
+                              scope as keyof typeof FEATURED_NETWORKS
+                            ]
+                          } (${scope})`
+                        : scope}
+                    </h3>
+
+                    <select
+                      className="accounts-select"
+                      value={selectedAccounts[scope] ?? ''}
+                      onChange={(evt) => {
+                        const newAddress = evt.target.value;
+                        setSelectedAccounts((prev) => ({
+                          ...prev,
+                          [scope]: newAddress,
+                        }));
+
+                        // modify the method request if it's a signing method (inject selected address)
+                        const currentMethod = selectedMethods[scope];
+                        if (currentMethod && currentMethod in SIGNING_METHODS) {
+                          const example = metamaskOpenrpcDocument?.methods.find(
+                            (method) =>
+                              (method as MethodObject).name === currentMethod,
+                          );
+
+                          if (example) {
+                            let exampleParams: JsonValue = openRPCExampleToJSON(
+                              example as MethodObject,
+                            );
+
+                            // inject the newly selected address
+                            exampleParams = insertSigningAddress(
+                              currentMethod,
+                              exampleParams,
+                              newAddress,
+                            );
+
+                            const updatedRequest = {
+                              method: 'wallet_invokeMethod',
+                              params: {
+                                scope,
+                                request: exampleParams,
+                              },
+                            };
+
+                            setInvokeMethodRequests((prev) => ({
+                              ...prev,
+                              [scope]: JSON.stringify(updatedRequest, null, 2),
+                            }));
+                          }
+                        }
+                      }}
+                    >
+                      <option value="">Select an account</option>
+                      {(details.accounts ?? []).map((account: string) => {
+                        const address = account.split(':').pop();
```

Comment:
feels a bit odd to do it like this since this should be a well-formed CAIP-10 account address. Thoughts on using [utils](https://github.com/MetaMask/utils/blob/main/src/caip-types.ts#L191) here instead?